### PR TITLE
Add gRPC support

### PR DESCRIPTION
**Pull request by ZachGarcia42**

### CODE REVIEW

**Comment by ZachGarcia42** [comment]
* **#39** <a href="https://app.graphite.dev/github/pr/cornellrocketryteam/Ground-Software/39?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ZachGarcia42 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: ground-server/app/grpc/page.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+useEffect(() => {
+    runSendCommand().then(res => {
+        try {
+            if(res.ack) {
+                setMessage("Command Acknowledged!")
+            }
+        } catch (err) {
+            setMessage("No Acknowledgement Received")
+        }
```
Comment (maxslarsson) [inline ground-server/app/grpc/page.js:17]: The `then` function on a promise only gets called if runSendCommand is successful, so I don't think the catch block will be called. If you do something like:

```javascript
runSendCommand().then(res => {
    setMessage("Command Acknowledged!")
}).catch(err => {
    setMessage("No Acknowledgement Received")
})
```

That should catch the error properly. You can also keep the `res.ack` if statement in if you want.

**Review by maxslarsson — CHANGES_REQUESTED** [review]

File: ground-server/app/grpc/page.js (hunk 1 above)
Comment (ZachGarcia42) [reply ground-server/app/grpc/page.js]: Thanks for catching this!

**Review by maxslarsson — APPROVED** [review]
Looks great! Thanks for implementing this!

**Comment by ZachGarcia42** [comment]
### Merge activity

* **Sep 22, 1:39 AM EDT**: @ZachGarcia42 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/cornellrocketryteam/Ground-Software/39).

**Pull request merged**